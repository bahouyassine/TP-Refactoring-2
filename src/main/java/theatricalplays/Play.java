package theatricalplays;

public class Play {

  public String name;
  public String type;

  public Play(String name, String type) {
    this.name = name;
    this.type = type;
  }

  public static int CaluclateTragedyPlayAmount(int audience){
    int thisAmount = 40000;
    if (audience > 30) {
      thisAmount += 1000 * (audience - 30);
    }
    return thisAmount;
  }
  public static int CaluclateComedyPlayAmount(int audience){
    int thisAmount = 30000;
    if (audience > 20) {
      thisAmount += 10000 + 500 * (audience - 20);
    }
    thisAmount += 300 * audience;
    return thisAmount;
  }

  public int CaluclatePlayAmount(int audience){
    int thisAmount = 0;
    switch (this.type) {
      case "tragedy":
        thisAmount = Play.CaluclateTragedyPlayAmount(audience);
        break;
      case "comedy":
        thisAmount = Play.CaluclateComedyPlayAmount(audience);
        break;
      default:
        throw new Error("unknown type: ${play.type}");
    }
    return thisAmount;
  }
}
