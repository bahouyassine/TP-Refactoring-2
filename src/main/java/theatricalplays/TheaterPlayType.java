package theatricalplays;

public enum TheaterPlayType {
    TRAGEDY, COMEDY, HISTORY, PASTORAL
  }
