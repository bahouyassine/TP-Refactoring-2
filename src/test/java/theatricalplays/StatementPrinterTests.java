package theatricalplays;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import theatricalplays.Play.TheaterPlayType;

import java.util.List;
import java.util.Map;

import static org.approvaltests.Approvals.verify;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {
        Map<String, Play> plays = Map.of(
                "hamlet",  new Play("Hamlet", TheaterPlayType.TRAGEDY),
                "as-like", new Play("As You Like It", TheaterPlayType.COMEDY),
                "othello", new Play("Othello", TheaterPlayType.TRAGEDY));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.toText2(invoice, plays);
        verify(result);
    }

    @Test
    void statementWithNewPlayTypes() {
        Map<String, Play> plays = Map.of(
                "henry-v",  new Play("Henry V", TheaterPlayType.HISTORY),
                "as-like", new Play("As You Like It", TheaterPlayType.PASTORAL));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("henry-v", 53),
                new Performance("as-like", 55)));
        StatementPrinter statementPrinter = new StatementPrinter();
        Assertions.assertThrows(Error.class, () -> {
            statementPrinter.toText2(invoice, plays);

        });
    }
}
