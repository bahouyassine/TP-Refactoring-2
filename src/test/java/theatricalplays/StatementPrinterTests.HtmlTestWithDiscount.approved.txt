<!DOCTYPE html><html>  <head>    <style>      body {        font-family: Arial, sans-serif;      }      h2 {        text-align: center;      }      table {        border-collapse: collapse;        width: 100%;      }      th,      td {        border: 1px solid black;        padding: 8px;        text-align: left;      }      th {        background-color: #f2f2f2;      }      ul {        list-style-type: square;        margin-left: 20px;      }      p {        margin: 10px 0;      }      .total-row {        font-weight: bold;      }    </style>  </head>  <body>    <h2>Invoice</h2>    <ul>      <li><strong>Client:</strong> BigCo</li>    </ul>    <table>      <tr>        <th>Piece</th>        <th>Seats Sold</th>        <th>Price</th>      </tr>      <tr>
<td>Hamlet</td>
<td>100</td>
<td>$1,100.00</td>
</tr>
<tr>
<td>As You Like It</td>
<td>120</td>
<td>$1,260.00</td>
</tr>
<tr>
<td>Othello</td>
<td>90</td>
<td>$1,000.00</td>
</tr>
      <tr class="total-row">        <td colspan="2">Total Owed:</td>        <td>3360.0</td>      </tr>      <tr class="total-row">        <td colspan="2">Fidelity Points Earned:</td>        <td>244</td>      </tr>    </table>    <div  style="    background-color: #f0f0f0;    padding: 20px;    border: 1px solid #ddd;    border-radius: 5px;  ">  <h2>Congratulations! You've earned a loyalty discount</h2>  <p>    A loyalty discount of    <span style="color: #0070ba; font-weight: bold">â‚¬15</span> has been applied.  </p>  <p>    <strong>Amount owed after discount:</strong>    <span style="color: #0070ba">$3,345.00</span>  </p>  <p>    <strong>Credits left after discount:</strong>    <span style="color: #0070ba">94 Credits</span>  </p></div>    <p>      Payment is required within 30 days. We can break your knees if you don't      do so !!!    </p>  </body></html>